### PR TITLE
Mathis/input restrict

### LINKED ISSUE #114: Restrict Input Values
On our buying and selling forms, the input fields should not allow the user to type in letters and random characters. We should use a key event (press, down, up) and prevent the invalid characters from appearing in the field. The same is true for repeated use of the decimal character, for example. There may be an existing React component that we can use for this.

### PR DESCRIPTION
**Pull request by cipherzzz**
**Issue:**
resolves #114 

**Summary**:
- Regex allows only [0-9] and .
- Allows 18 digits to the left and 6-18 to the right of the decimal - depending on the coin.
- The current formatting does not allow more than 6 decimals and I didn't want to wade into that
- I did not see any `tests` in the dApp so I didn't add one for the regex

### CODE REVIEW

**Review by sparrowDom — COMMENTED** [review]
A few minor comments otherwise great job :+1:

File: dapp/src/utils/math.js [hunk 1]
```diff
@@ -96,3 +96,30 @@ export async function displayCurrency(balance, contract) {
...
+  let decimals = 18
+
+  switch (coin) {
+    case 'usdc':
+      decimals = 6
+      break
+    case 'usdt':
+      decimals = 6
+      break
+    default:
```
[the remark this answers is not in the document]
Comment (cipherzzz) [reply dapp/src/utils/math.js:115]: @sparrowDom - Sounds good. I'll add this suggestion and the one below! 💯

File: dapp/src/utils/math.js [hunk 2]
```diff
@@ -96,3 +96,30 @@ export async function displayCurrency(balance, contract) {
...
+      decimals = 18
```
[the remark this answers is not in the document]
Comment (sparrowDom) [reply dapp/src/utils/math.js:116]: Ah this should be `throw new Error` my bad ;)

**Review by sparrowDom — APPROVED** [review]
Great work on this one @cipherzzz, thanks for doing this 💪

File: dapp/src/utils/math.js [hunk 3]
```diff
@@ -96,3 +96,32 @@ export async function displayCurrency(balance, contract) {
   if (!balance) return
   return ethers.utils.formatUnits(balance, await contract.decimals())
 }
+
+export function checkValidInputForCoin(amount, coin) {
+  if (amount === '') {
+    amount = '0.00'
+  }
+
+  const COIN = coin.toLowerCase()
```
Comment (franckc) [inline dapp/src/utils/math.js:105]: Sorry for being late to the review party.
This constant is not used anywhere? It should be used in the switch statement below.

Also I think our coding style tens to use all capital for hardcoded constants and camel case for everything else. So perhaps rename COIN to coinSymbol or equivalent.

**Pull request merged**